src/main/Extensions/Base64Conversion/Base64Conversion.ts

```diff
@@ -18,8 +33,69 @@ export class Base64Conversion implements Extension {
     public constructor(
         private readonly assetPathResolver: AssetPathResolver,
         private readonly translator: Translator,
+        private readonly settingsManager: SettingsManager,
     ) {}
 
+    public getInstantSearchResultItems(searchTerm: string): InstantSearchResultItems {
+        const encodeDecodePrefix: string = this.getSettingValue("encodeDecodePrefix");
+        const encodePrefix: string = this.getSettingValue("encodePrefix");
+        const decodePrefix: string = this.getSettingValue("decodePrefix");
```

Comment:
I suggest doing this little typescript trick:

Adjust these two methods to the following:

```typescript
public getSettingDefaultValue<T extends keyof Settings>(key: T): Settings[T] {
    return this.defaultSettings[key];
}

private getSettingValue<T extends keyof Settings>(key: T): Settings[T] {
    return this.settingsManager.getValue(getExtensionSettingKey(this.id, key), this.getSettingDefaultValue(key));
}
```

Typescript will now know what the type will be based on the type definition in `Base64ConversionSettings` so you can remove the manual type hints here:

```suggestion
        const encodeDecodePrefix = this.getSettingValue("encodeDecodePrefix");
        const encodePrefix = this.getSettingValue("encodePrefix");
        const decodePrefix = this.getSettingValue("decodePrefix");
```